lib/app/components/pages/chat_room/views/chat_room_view.dart

```diff
@@ -1,68 +1,125 @@
+                          )
+                        : ReceivedMessage(
+                            profileImage: profileImage,
+                            senderName: senderName,
+                            text: message,
+                          ),
                     const SizedBox(
-                      height: 20,
+                      height: 10,
                     ),
-                  controller.isSender(senderId)
-                      ? SentMessage(
-                          text: message,
-                        )
-                      : ReceivedMessage(
-                          profileImage: profileImage,
-                          senderName: senderName,
-                          text: message,
-                        ),
-                  const SizedBox(
-                    height: 10,
-                  ),
-                ],
-              );
-            },
-          ),
-          const MessageBar(),
-        ],
+                  ],
+                );
+              },
+            ),
+            MessageBar(
+              controller: controller.messageTextController.value,
+              onCameraPressed: () {}, //TODO: カメラ機能は別途実装
+              onPhotoPressed: () {}, //TODO: 画像追加機能は別途実装
+              onSendPressed: _onSendPressed, //TODO: 送信機能は別途実装
+            ),
+          ],
+        ),
       ),
     );
   }
+
+  void _onSendPressed() {
+    final senderId = controller.authService.uid.val;
+    final senderName = controller.authService.userName.val;
+    final profileImage = controller.authService.profileImage.val;
+    final messageText = controller.messageTextController.value.text;
+
+    final message = ChatMessageModel(
+      senderId: senderId,
+      senderName: senderName,
+      profileImage: profileImage,
+      media: null, // TODO: メディア追加機能作成時にテキストと画像を同時に送れるようにする？
+      message: messageText,
+      createdAt: FieldValue.serverTimestamp(),
+    );
+
+    _insertMessage(message);
+
+    // 入力欄を初期化
+    controller.messageTextController.value.text = '';
+  }
+
+  // TODO: firebase接続後はこの関数を使用　動作未確認
+  // void _insertMessage(ChatMessageModel message) async {
+  //   final roomId = controller.chatService.chatRoom.roomId;
+  //   final content = message.toJson();
+  //   await FirebaseFirestore.instance
+  //       .collection('chatRooms')
+  //       .doc(roomId)
+  //       .update({
+  //     'message': FieldValue.arrayUnion([content]),
+  //   });
+  // }
+
+  // TODO: firebase接続後はこの関数は削除
+  void _insertMessage(ChatMessageModel message) {
+    final roomId = controller.chatService.chatRoom.roomId;
+    final content = message.toJson();
+
+    // 対象メッセージのListが含まれるchatRoomDataの要素のindex
+    final targetIndex =
+        controller.chatRooms.indexWhere((element) => element.roomId == roomId);
+
+    // 追加する対象メッセージのList
+    final targetMessageList = controller.chatRooms[targetIndex].messages;
+    targetMessageList.insert(
+        targetMessageList.length, content.cast<String, String?>());
```

Comment:
RxList型を追加したら`refresh()`を呼ばないと反映されないので以下を追加しましょう。
```suggestion
    targetMessageList.insert(
        targetMessageList.length, content.cast<String, String?>());
    controller.chatService.chatRoom.messages.refresh();
```

Reply:
@king-kazu39 
自分で試行錯誤する中でcontroller.update()は使ってみたんですが、うまくいかず困っていました。ありがとうございます。